### ISSUE TITLE
Equivalent to NASM %defstr

### ISSUE THREAD

**Issue by travisdowns** [yasm/yasm#95]
This is either a question and if the answer is "no equivalent" a request for enhancement.

When compiling with `nasm` I used the `%defstr` macro to turn an (unquoted) macro parameter into a string, e.g., so I can define a string constant with that value in `.rodata`. Example:

```
%macro abi_checked_function 1
GLOBAL %1:function
%1:

; some code omitted here

; load the function name (ok to clobber rdi since it's callee-saved)
mov rdi, %1_thunk_fn_name

; here we store strings needed by the failure cases, in the .rodata section
[section .rodata]
%1_thunk_fn_name:
%defstr fname %1
db fname,0

; restore the previous section
__SECT__

%1_inner:
%endmacro
```

Is there any equivalent in in yasm for this pattern?

**Comment by PeterJohnson** [comment]
Since yasm uses the nasm preprocessor, the answer is `%defstr`.  But you're right, it doesn't support it today.  The solution will be to import a more recent version of the nasm preprocessor code into modules/preprocs/nasm.

**Comment by travisdowns** [comment]
I guess I can use the `nasm` pre-processor with `nasm -E` (pre-process only) and then send the output through `yasm`, e.g., because I want to use `yasm` because it doesn't generate broken dwarf info.

**Comment by PeterJohnson** [comment]
Yes, that will work too.  You can even pipe it (since `yasm -` accepts input from stdin, you can do `nasm -E in.asm | yasm -felf32 -o in.o -` or similar.

**Comment by travisdowns** [comment]
@PeterJohnson - thanks. FWIW I tried the idea of using `nasm` as a pre-processor for `yasm`, but it failed because `ALIGN 16` in nasm turns into a `sectalign` directive, which apparently yasm doesn't support.
